src/main/java/org/opensearch/ad/feature/SearchFeatureDao.java

```diff
@@ -496,6 +496,24 @@ public void getEntityMinDataTime(AnomalyDetector detector, Entity entity, Action
             );
     }
 
+    /**
+     * Get the entity's earliest timestamps
+     * @param detector detector config
+     * @param listener listener to return back the requested timestamps
+     */
+    public void getMinDataTime(AnomalyDetector detector, ActionListener<Optional<Long>> listener) {
```

Comment:
Shouldn't the minimum time be related to what the specified interval is? Why not just default to using a min time that's equal to the min time that would be used during cold start?

Reply:
I chose to use minimum data time to get the the top entity so it encapsulates all data that could be found with the different interval recommendation. To decrease amount of queries being made which are already quite high I didn't want to get the top entity based on each interval I try again and again. I could however change this to get the minimum time based on the longest possible interval that will be tried in the recommendation (which is currently 60 minutes). I think I will change to that if that makes sense to you

Reply:
> I could however change this to get the minimum time based on the longest possible interval that will be tried in the recommendation (which is currently 60 minutes)

I think this is better than the current soln, yes. Even so, why not just default to using whatever the min timestamp would be as a lower bounds, instead of needing to make a separate query to see what the actual min is in the index?